pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -403,6 +403,13 @@ func (f *ring1Factory) AttachablePodForObject(object runtime.Object, timeout tim
 			return nil, fmt.Errorf("invalid label selector: %v", err)
 		}
 
+	case *api.Service:
```

Comment:
can we move this change out of this PR to a follow-up that includes the target port mapping discussed in https://github.com/kubernetes/kubernetes/issues/59733

Reply:
I can do that.